spec/models/article_spec.rb

```diff
@@ -681,6 +681,38 @@ def build_and_validate_article(*args)
     end
   end
 
+  describe ".active_threads" do
+    let!(:filtered_article) do
+      create(:article, user: user, score: -2, tags: "discuss, watercooler")
+    end
+
+    context "when the articles fall within the constraints" do
+      it "returns the latest published article within the score constraints" do
+        articles = described_class.active_threads("discuss", "latest", 10)
+        expect(articles.first[0]).to eq(filtered_article.path)
+      end
+
+      it "returns the published article within the time_ago and score constraints" do
+        articles = described_class.active_threads("discuss", 1.hour.ago, 10)
+        expect(articles.first[0]).to eq(filtered_article.path)
+      end
+
+      it "returns the published article within the published_at and score constraints" do
+        articles = described_class.active_threads("discuss", 6.days.ago, 10)
+        expect(articles.first[0]).to eq(filtered_article.path)
+      end
+    end
+
+    context "when the articles do not fall within the constraints" do
+      it "returns the published article with the corresponding tag even if it does not fall within the constraints" do
+        article.update_columns(score: -25, cached_tag_list: "discuss")
+        articles = described_class.active_threads("discuss", nil, 10)
+        expect(articles.first[0]).to eq(filtered_article.path)
+        article.update_columns(score: -2, cached_tag_list: "")
```

Comment:
This setup (updating the necessary columns on `article) is necessary to ensure that I both get the correct article back and that the setup of future tests remains as expected!

Reply:
Would it maybe be easier to just create a different article local to this spec?

Reply:
🤔  I'm running into the same issue in the new spec that I created for `Articles::ActiveThreadsQuery` which makes me think that this is an actual issue. Additionally, I am unable to see my lower-quality articles in the homepage sidebar anymore, which is alarming. Yesterday evening I reset my DB so that I could manually QA my work from a clean slate (like a reviewer would and since I had to `rebase` because my branch was out of date anyway) while creating this PR and haven't been able to see any lower-quality articles in the sidebar since. I _swear_ that they were showing up prior to that. Some of the articles that were previously showing up had scores of `-25` and the articles themselves did not have any reactions, comments, etc. -- the articles were mainly just the "latest" articles for the tag.

 I'm experiencing this issue now, too, which is why the new spec, `active_threads_query_spec.rb` is failing. That said, I have a hunch that it's something to do with the `score` (when I alter the `score` in the spec to fall _within_ the query's `MINIMUM_SCORE`, the test passes) and/or that the lower-quality articles are not coming through in the query itself (probably due to the `score` constraint being met) -- there are high-quality articles that exist, so we never hit the `||` in the query, thus the lower-quality articles never show up.

I'm digging into this issue now and will continue on with it tomorrow if I don't resolve it. 🤔